Assets/Scripts/Other/Battle.cs

```diff
@@ -0,0 +1,63 @@
+﻿
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Battle : MonoBehaviour
+{
+  public bool isFighting=false;
```

Comment:
Try to use proper access modifiers. In this case `isFighting` should not be exposed to any class outside and should be private.